code-challenges.test.js

```diff
@@ -13,40 +13,128 @@
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+describe("capArray", () => {
+  it("returns an array with a sentence about each person with their name capitalized.", () => {
+    const hitchhikersCharacters = [
+      { name: "ford prefect", occupation: "a hitchhiker" },
+      { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+      { name: "arthur dent", occupation: "a radio employee" }
+    ]
+    // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+    expect(capArray(hitchhikersCharacters)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
+//good failure: ReferenceError: CapArray is not defined
 
 // b) Create the function that makes the test pass.
-
 // Pseudo code:
+// function name: CapArray
+// input: array of objects
+// output: an array of strings
+// process: create a function with a parameter of array, use map method to iterate through all the key:value pairs inside the array, create a variable for name to split the values of names so its easier to find the index of the last name, create a variable for capitalizing the first letter of the first name, seperate the names to have first name in its own array and last name in its own array, then create a variable to uppercase the last name, use slice to add the capitalized first letter to the rest of the first name and same for the last name, then use the concat method to add the first and last names together, the return the string interpulate line. 
+
+
+const capArray = (array) => {
+  return array.map((value) => {
+    const name = value.name.split('').indexOf(' ') // Finds the indexOf where the space is in each name. 
+    const firstCase = value.name.charAt(0).toUpperCase() //Capitalizes the first letter of the first name
+    const seperateName = value.name.split(' ') //Makes the first name be in the 0 index while the last name is in the 1st index. 
+    const firstName = seperateName[0] // Holds just the first index so just the first name.
+    const lastCase = value.name.charAt(name + 1).toUpperCase() //Capitalizes the first letter of the last name
+    const lastName = seperateName[1] // Holds just the last index so just the last name.
+    const capFirstName = firstCase + firstName.slice(1) //Concats the capital letter of the first name with everything sliced after the index of 1, including 1. 
+    const capLastName = lastCase + lastName.slice(1) //Concats the capital letter of the last name with everything sliced after the index of 1, including 1. 
+    const fullName = capFirstName + ' ' + capLastName
+    return `${fullName} is ${value.occupation}.`
```

Comment:
I like the step by step logic here. I suggest a refactor to eliminate the amount of variables created.

I also saw your comment about wanting to see different approaches, here's one:

const capArray = (arr) => {
   return arr.map(obj => `${obj.name.replace(/\b\w/g, match => match.toUpperCase())} is ${obj.occupation}.`)
}

There are backticks in front of the $ and after the period.

This uses regex with the `.replace` method to target the first letter of each word in the name string and capitalize it.